src/core/thread/mle.cpp

```diff
@@ -1792,6 +1792,8 @@ ThreadError Mle::HandleParentResponse(const Message &aMessage, const Ip6::Messag
 
     otLogInfoMle("Received Parent Response\n");
 
+    assert(threadMessageInfo != NULL);
```

Comment:
I think we should just add the `assert()` statement to the top of `Mle::HandleUdpReceive()`.  We can cover both assert checks with a single top-level one.